tests/Tests/Axis3d.elm

```diff
@@ -133,6 +136,65 @@ intersectionWithPlane =
         )
 
 
+intersectionWithTriangle : Test
+intersectionWithTriangle =
+    Test.fuzz3
+        Fuzz.triangle3d
+        Fuzz.direction3d
+        Fuzz.vector3d
+        "intersectionWithTriangle works properly"
+        (\triangle axisDirection parameters ->
+            let
+                ( p0, p1, p2 ) =
+                    Triangle3d.vertices triangle
+
+                e1 =
+                    Vector3d.from p0 p1
+
+                e2 =
+                    Vector3d.from p0 p2
+
+                ( u, v, t ) =
```

Comment:
I think I'd be inclined to use something like 

```elm
Fuzz.tuple3
    ( Fuzz.floatRange -0.5 1.5
    , Fuzz.floatRange -0.5 1.5
    , Fuzz.floatRange -0.5 1.5
    )
```

to generate a `( Float, Float, Float )` instead of using `Fuzz.vector3d` - `Fuzz.vector3d` will generate vectors with components in the -10..10 range so _most_ of the values you generate will be out of range, very few tests will actually trigger an intersection
```

Reply:
Yes, great idea! This now also produces some degenerate examples that I will need to handle in the test code. Which is a good thing, obviously.